components/event-bus/cmd/event-bus-publish-knative/handlers/publish.go

```diff
@@ -80,7 +77,13 @@ func handleKnativePublishRequest(w http.ResponseWriter, r *http.Request, knative
 
 	// generate event-id if there is none
 	if len(publishRequest.EventID) == 0 {
-		publishRequest.EventID = generateEventID()
+		eventID, errEventID := generateEventID()
+		if errEventID != nil {
+			err = api.ErrorResponseInternalServer()
+			log.Printf("EventID generation failed: %v", err)
+			publish.SendJSONError(w, err)
+		}
```

Comment:
Please put a return statement here, because at this point, the publish request should fail.

Reply:
Good catch 👍